huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_mysql_account_privilege.go

```diff
@@ -0,0 +1,479 @@
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceGaussDBAccountPrivilege() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceGaussDBAccountPrivilegeCreate,
+		ReadContext:   resourceGaussDBAccountPrivilegeRead,
+		DeleteContext: resourceGaussDBAccountPrivilegeDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"instance_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the ID of the GaussDB MySQL instance.`,
+			},
+			"account_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the database username.`,
+			},
+			"host": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the host IP address.`,
+			},
+			"databases": {
+				Type:        schema.TypeList,
+				Elem:        AccountPrivilegeDatabaseSchema(),
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the list of the databases.`,
+			},
+		},
+	}
+}
+
+func AccountPrivilegeDatabaseSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the database name.`,
+			},
+			"readonly": {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: `Specifies whether the database permission is read-only.`,
+			},
+		},
+	}
+	return &sc
+}
+
+func resourceGaussDBAccountPrivilegeCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// createGaussDBAccount: create a GaussDB MySQL account privilege
+	var (
+		createGaussDBAccountPrivilegeHttpUrl = "v3/{project_id}/instances/{instance_id}/db-users/privilege"
+		createGaussDBAccountPrivilegeProduct = "gaussdb"
+	)
+	createGaussDBAccountPrivilegeClient, err := cfg.NewServiceClient(createGaussDBAccountPrivilegeProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating GaussDB Client: %s", err)
+	}
+
+	instanceID := d.Get("instance_id").(string)
+	accountName := d.Get("account_name").(string)
+	host := "%"
+	if h := d.Get("host").(string); h != "" {
+		host = h
+	}
```

Comment:
host is required, so it won't be empty, why we need this?

Reply:
it is not needed, so it has been deleted